### ISSUE TITLE
[Accessibility] "All of DSpace" menu in header does not expose the presence of the submenu & its state via ARIA attributes

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#2699]
Originally reported in #1174 via the Deque Analysis in 2021. Rated "serious" by Deque. This is one level less than the highest "critical" rating, and is described by Deque as follows:
> Serious: This issue results in serious barriers for people with disabilities, and will partially prevent them from accessing fundamental features or content. People relying on assistive technologies will experience significant frustration as a result. Issues falling under this category are major problems, and remediation should be a priority.

**Describe the bug**

(Deque Issue ID: 468467) "All of DSpace" submenu: The presence of the submenu and its state are not exposed.
* Suggested fixes include BOTH of these: 
    * (1) Use `aria-haspopup="menu"` on the element that toggles the submenu open/closed. AND 
    * (2) Use the `aria-expanded` attribute on the element that toggles the submenu open/closed with a value of "true" (when the submenu is expanded) or "false" (when the submenu is collapsed).

**To Reproduce**
1. Can be seen on the https://demo.dspace.org site.  These ARIA labels are missing and never updated based on the current state of the menu.

**Related work**
Issue initially reported in #1174

**Comment by tdonohue** [comment]
Assigned to @davide-negretti as his larger refactor in #2676 fixes this small issue.

**Comment by tdonohue** [comment]
Ported to 7.x in #2858
